### PR TITLE
fix(rpc module): unsubscribe according ethereum pubsub spec

### PR DESCRIPTION
**Pull request by niklasad1**
As we already have adopted the [ethereum pubsub spec](https://geth.ethereum.org/docs/rpc/pubsub) for subscriptions let's stick to it completely.

### CODE REVIEW

File: core/src/server/rpc_module.rs [hunk 1]
```diff
@@ -681,27 +680,17 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
...
-					if subscribers
+					let result = subscribers
 						.lock()
 						.remove(&SubscriptionKey { conn_id: conn.conn_id, sub_id: sub_id.clone() })
-						.is_some()
-					{
-						sink.send_response(id, "Unsubscribed")
-					} else {
-						let err = to_json_raw_value(&format!(
-							"Invalid subscription ID={}",
```
Comment (dvdplm) [inline core/src/server/rpc_module.rs:699]: The only downside to this PR is that we loose this information (I think?). 

With this PR, when an unsubscribe call fails, what do users get back?

**Review by dvdplm — APPROVED** [review]

File: core/src/server/rpc_module.rs (hunk 1 above)
Comment (niklasad1) [reply core/src/server/rpc_module.rs:699]: yeah, unfortunately they will just receive a `bool` set to false and then the client(s) have to figure it on their own :(

Comment (dvdplm) [reply core/src/server/rpc_module.rs:699]: Let's document this properly in the macro docs (and on this method): "When attempting to unsubscribe with an unknown subscription ID this call will return `false`" or something along those lines?

Comment (niklasad1) [reply core/src/server/rpc_module.rs:699]: I guess it's better to document that we implement the `ethereum pubsub specification` when we refer to subscriptions and it's possible to provide custom subscription IDs but that's it.

Sure, we could add some documentation to both `RpcModule::register_subscription` and the macros.

**Review by jsdw — APPROVED** [review]

**Pull request merged**